### PR TITLE
[PFsense] Add PFsense Integration

### LINKED ISSUE #1206: pfSense
### Description
pfSense is a free, open source customized distribution of FreeBSD specifically tailored for use as a firewall and router that is entirely managed via web interface. In addition to being a powerful, flexible firewalling and routing platform, it includes a long list of related features and a package system allowing further expandability without adding bloat and potential security vulnerabilities to the base distribution.

Draft PR from @legoguy1000 available here: https://github.com/elastic/integrations/pull/1286

### Architecture
Syslog (both RFC3164 and RFC5424) is supported - relevant documentation [here](https://docs.netgate.com/pfsense/en/latest/monitoring/logs/remote.html). 

# Integration release checklist

This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes

- [ ] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md)
- [ ] Supported versions of the monitoring target are documented
- [ ] Supported operating systems are documented (if applicable)
- [ ] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [ ] Documentation exists
- [ ] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [ ] At least a manual test with ES / Kibana / Agent has been performed.
- [ ] Required Kibana version set to:

### New Package

- [ ] Screenshot of the "Add Integration" page on Fleet added

### Dashboards changes

- [ ] Dashboards exists
- [ ] Screenshots added or updated
- [ ] Datastream filters added to visualizations

### Log dataset changes

- [ ] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [ ] Generated output for at least 1 log file exists
- [ ] Sample event (`sample_event.json`) exists

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Add PFsense Integration.  Currently parses Firewall (filterlog), Open VPN (openvpn), IPSEC (charon), DHCP (dhcpd), DNS (unbound), HAProxy (haproxy) logs

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Author's Checklist

- [ ]

## How to test this PR locally

```
cd integrations/packages/pfsense
elastic-package check && elastic-package stack down && elastic-package build && elastic-package stack up -d -v -p default && eval "$(elastic-package stack shellinit)" && elastic-package test  -v
```

## Related issues

- Closes #1206

### CODE REVIEW

**Comment by P1llus** [comment]
Run tests

**Comment by P1llus** [comment]
run tests

**Comment by P1llus** [comment]
Run tests

**Comment by P1llus** [comment]
Run tests

File: packages/pfsense/_dev/build/build.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+dependencies:
+  ecs:
+    reference: git@1.11
```
Comment (legoguy1000) [inline packages/pfsense/_dev/build/build.yml:3]: Need to update to 1.12

File: packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,170 @@
+---
+description: Pipeline for PFsense
+processors:
+  - set:
+      field: event.ingested
+      value: '{{_ingest.timestamp}}'
+  - set:
```
Comment (legoguy1000) [inline packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/default.yml:7]: Need to update to 1.12

**Review by P1llus — APPROVED** [review]
@legoguy1000 Great work! Can you confirm you also tested to adding the package to a profile of some sort? I am just curious about the visualizations

**Comment by legoguy1000** [comment]
> @legoguy1000 Great work! Can you confirm you also tested to adding the package to a profile of some sort? I am just curious about the visualizations

I did and I can test them again before merging.  The only ones that were iffy were the Vega visualizations. These visualizations were ones I used for myself that I got from https://github.com/pfelk/pfelk/tree/main/etc/pfelk/dashboard.  I also just realized that the dashboards images have my own domain littered throughout so I'm going to make now images without them.

**Comment by legoguy1000** [comment]
I found some grok patterns that don't match.  Updates to come.

**Comment by legoguy1000** [comment]
After adding the timezone comfig, I'm getting some errors i've never seen before.
```
pfsense/log test-pfsense-bsd.log:
[0] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-07-03T19:10:30.000-04:00
[1] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-07-03T19:10:31.000-04:00
[2] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-07-03T19:10:32.000-04:00
[3] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-07-03T19:10:33.000-04:00
[4] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-07-04T11:10:45.000-04:00
[5] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-07-04T11:10:54.000-04:00
pfsense/log test-pfsense-dhcp.log:
[0] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-07-04T09:39:40.000-04:00
[1] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-07-04T09:39:41.000-04:00
pfsense/log test-pfsense-haproxy.log:
[0] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-08-15T16:15:10.549-04:00
[1] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-08-15T16:15:18.407-04:00
[2] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-08-15T16:15:18.502-04:00
pfsense/log test-pfsense-openvpn.log:
[0] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-07-03T21:42:57.000-04:00
pfsense/log test-pfsense-unbound.log:
[0] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-08-15T16:18:59.000-04:00
[1] dynamic field "@timestamp" doesn't match the pattern (^[0-9]{4}(-[0-9]{2}){2}T[0-9]{2}(:[0-9]{2}){2}\.[0-9]{3}Z$): 2021-08-15T16:19:02.000-04:00
```

**Comment by P1llus** [comment]
That is related to this line here:
https://github.com/elastic/integrations/pull/1286/files#diff-c48a9c982d57fcfb8c046030abcc8c988d3f5ce2f1ecccda9d76004539b33181R3

**Pull request merged**